src/theme/types.ts

```diff
@@ -0,0 +1,218 @@
+    "border-primary": string | ((props: any) => string);
+    "border-primary-subtle": string | ((props: any) => string);
+    "border-hover-strong": string | ((props: any) => string);
+    "border-selected-subtle": string | ((props: any) => string);
+    "border-selected-subtlest": string | ((props: any) => string);
+    "border-focus": string | ((props: any) => string);
+    "border-focus-strong": string | ((props: any) => string);
+    "border-selected-disabled": string | ((props: any) => string);
+    "border-warning": string | ((props: any) => string);
+    "border-error": string | ((props: any) => string);
+    "border-error-focus": string | ((props: any) => string);
+    "border-info": string | ((props: any) => string);
+
+    background: string | ((props: any) => string);
+    "background-hover": string | ((props: any) => string);
+    "background-selected": string | ((props: any) => string);
+    "background-disabled": string | ((props: any) => string);
+    "background-success": string | ((props: any) => string);
+    "background-inverse": string | ((props: any) => string);
+    "background-subtle": string | ((props: any) => string);
+    "background-hover-strong": string | ((props: any) => string);
+    "background-hover-subtle": string | ((props: any) => string);
+    "background-hover-neutral": string | ((props: any) => string);
+    "background-primary": string | ((props: any) => string);
+    "background-primary-hover": string | ((props: any) => string);
+    "background-primary-subtle": string | ((props: any) => string);
+    "background-primary-subtlest": string | ((props: any) => string);
+    "background-primary-subtlest-hover": string | ((props: any) => string);
+    "background-primary-subtlest-selected": string | ((props: any) => string);
+    "background-selected-strong": string | ((props: any) => string);
+    "background-selected-hover": string | ((props: any) => string);
+    "background-selected-disabled": string | ((props: any) => string);
+    "background-warning": string | ((props: any) => string);
+    "background-error": string | ((props: any) => string);
+    "background-info": string | ((props: any) => string);
+
+    "overlay-strong": string | ((props: any) => string);
+    "overlay-subtle": string | ((props: any) => string);
+    "overlay-inverse-gradient": string | ((props: any) => string);
+
+    hyperlink: string | ((props: any) => string);
+    "hyperlink-inverse": string | ((props: any) => string);
+    "focus-ring": string | ((props: any) => string);
+    "focus-ring-inverse": string | ((props: any) => string);
+};
+
+export type ColourScheme =
+    | "lifesg"
+    | "bookingsg"
+    | "rbs"
+    | "mylegacy"
+    | "ccube";
+
+export type ColourCollectionsMap = {
+    [key in ColourScheme]: ColourSet;
+};
+
+export type SematicColourCollectionMap = {
+    [key in ColourScheme]: SematicColourSet;
+};
+
+export enum ThemeContextKeys {
+    colourScheme = "colourScheme",
+}
```

Comment:
wondering if this enum is still required, or if the keys can be hardcoded

Reply:
Yea i dont see a need to define it that way it can be hardcoded shall I change it to be hardcoded?